src/main.cpp

```diff
@@ -2242,6 +2176,15 @@ bool CheckTxInputs(const CTransactionBase& txBase, CValidationState& state, cons
                 REJECT_INVALID, "bad-txns-inputvalues-outofrange");
     }
 
+    try {
+        nValueIn += txBase.GetCSWValueIn();
+        if (!MoneyRange(nValueIn))
+            throw std::runtime_error("Total inputs value out of range.");
```

Comment:
We should not rethrow the std::runtime_error(), but should do return state.DoS instead as in the block below

Reply:
I do not see where the first runtime_error is thrown in the first place